### ISSUE TITLE
Fix 12 hour format for 12:00am

### ISSUE THREAD

**Issue by biondizzle** [atatanasov/gijgo#424]
Can someone edit this file:
https://github.com/atatanasov/gijgo/blob/master/src/timepicker/js/timepicker.base.js

change line 316 from   if (hour >= 12) { to   if (hour > 12) {

the >= makes the 12 a 0 which is incorrect for 12h format. I would do it but I am on an iPad at the moment lol.

**Comment by khaladj** [comment]
Same issue for me.

**Comment by cuartas15** [comment]
This problem still exists

**Comment by atatanasov** [comment]
Do you have this issue when you set the time through the code with value method or you get the value from the control. Could you provide replication steps.

**Comment by cuartas15** [comment]
it's very simple to replicate
If I choose this time:
![imagen](https://user-images.githubusercontent.com/14366008/52968636-d9d32c00-337b-11e9-8700-15b4f1162459.png)
I get this:
![imagen](https://user-images.githubusercontent.com/14366008/52968661-e6578480-337b-11e9-8201-22258165afbf.png)

Another issue is that if I choose a PM time
![imagen](https://user-images.githubusercontent.com/14366008/52968712-09823400-337c-11e9-99df-31efef141754.png)
And I want to edit such time again, the time is set to AM instead of PM as it should be
![imagen](https://user-images.githubusercontent.com/14366008/52968781-46e6c180-337c-11e9-9334-9a20e4bf794b.png)

Last issue, if I have 12PM
![imagen](https://user-images.githubusercontent.com/14366008/52968797-58c86480-337c-11e9-8626-afa9f19cae3b.png)
And I open the modal It looks like this:
![imagen](https://user-images.githubusercontent.com/14366008/52968822-71d11580-337c-11e9-83a8-3ad8b425e9a8.png)
if I press ok, it will interpret it like it was 12AM and then changes the time to 00:00:00AM again

**Comment by mcqueenaece** [comment]
Same problem here.  Displays correctly if value is set in the method.

**Comment by fernandogutierrez27** [comment]
Hi! I had a similar issue.

When I was picking a time for the first time and my local time was between 12PM and 13PM, the selected time on the picker was correctly displayed as 12:XX PM, but the saved time on the input was 00:XX. 

This happens only on the first time, if I open the picker again, now the displayed time is 00:XX AM.

I made a modification on the following line of the source code: https://github.com/atatanasov/gijgo/blob/1968d51097a82b44e78dfd70d03b9eb0bdb2e312/src/timepicker/js/timepicker.base.js#L607

Basically, the original code was:
```
$clock.attr('mode', hour > 12 ? 'pm' : 'am');
```
and the modified code is:
```
$clock.attr('mode', hour >= 12 ? 'pm' : 'am');
```
Problem solved.

I'll try to do a PR, I've never do it before, but I can learn.

**Comment by ZeroGravityWeb** [comment]
I am having the same issue as @cuartas15. If the time in the field is 3:45 PM and you open the picker it shows 3:45 AM. @fernandogutierrez27 fix won't work because the hour is always going to be less than 12 when set to 12 hour time.

These are the settings I have to initialize the picker.
```
var timepicker = $('#timepicker').timepicker({
	mode: 'ampm',
	format: 'h:MM TT'
});
```
Is there a fix for this?

**Comment by cuartas15** [comment]
I managed to fix most of the problems long time ago editing the source file but those fixes are too specific for my application

**Comment by ZeroGravityWeb** [comment]
Thanks for the quick response @cuartas15. Our client was saying times were being set to AM whenever they edited the time. We hunted our code but couldn't see any reason why it may be happening until I noticed this issue yesterday. I was hoping we wouldn't have to try and debug it ourselves. Sigh...

**Comment by cuartas15** [comment]
Yeah, this library doesn't have an active support, I really like it but the cold reality is that there may be better options out there.
Like I said, I had to go and learn the source code in order to fix the multiple issues (like the datetimepicker not having a 12h format time picker, etc)
